### DISCUSSION TITLE
Ansible Wifi_login_pages installing incompatible node (v16.20.2) and yarn (1.22.22)

### DISCUSSION THREAD

**Discussion by 3pleo** [openwisp/ansible-openwisp-wifi-login-pages#43] [Q&A]
Hi,
Trying to deploy ansible wifi-login-pages with following role on to target system running Ubuntu 22.04.5 LTS:
ansible-galaxy install openwisp.wifi_login_pages

Noticed it was deplying Node 16 `[TASK [openwisp.wifi_login_pages : Add nodejs 16.x ppa for apt repo]`, and threw following incompatible node module errors:

`FAILED! => {"changed": false, "cmd": "/usr/bin/yarn install --non-interactive", "msg": "error marked@12.0.0: The engine \"node\" is incompatible with this module. Expected version \">= 18\". Got \"16.20.2\"\nerror Found incompatible module.", "rc": 1, "stderr": "error marked@12.0.0: The engine \"node\" is incompatible with this module. Expected version \">= 18\". Got \"16.20.2\"\nerror Found incompatible module.\n", "stderr_lines": ["error marked@12.0.0: The engine \"node\" is incompatible with this module. Expected version \">= 18\". Got \"16.20.2\"", "error Found incompatible module."], "stdout": "yarn install v1.22.22\n[1/4] Resolving packages...\n[2/4] Fetching packages...\ninfo Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.\n", "stdout_lines": ["yarn install v1.22.22", "[1/4] Resolving packages...", "[2/4] Fetching packages...", "info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command."]}`

`FAILED! => {"changed": true, "cmd": ["yarn", "build"], "delta": "0:00:00.555864", "end": "2024-09-19 15:58:14.569398", "msg": "non-zero return code", "rc": 9, "start": "2024-09-19 15:58:14.013534", "stderr": "warning Cannot find a suitable global folder. Tried these: \"/usr/local, /var/www/.yarn\"\n/usr/bin/node: --openssl-legacy-provider is not allowed in NODE_OPTIONS\nerror Command failed with exit code 9.", "stderr_lines": ["warning Cannot find a suitable global folder. Tried these: \"/usr/local, /var/www/.yarn\"", "/usr/bin/node: --openssl-legacy-provider is not allowed in NODE_OPTIONS", "error Command failed with exit code 9."], "stdout": "yarn run v1.22.22\n$ export NODE_OPTIONS=--openssl-legacy-provider && npm run setup && webpack --mode production --progress --config config/webpack.config.js\ninfo Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.", "stdout_lines": ["yarn run v1.22.22", "$ export NODE_OPTIONS=--openssl-legacy-provider && npm run setup && webpack --mode production --progress --config config/webpack.config.js", "info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command."]}`

I'm I doing anything wrong?

Regards

**Comment by nemesifier** [comment]
See https://github.com/openwisp/ansible-openwisp-wifi-login-pages/commit/83b881882ba539677899744af3ccf30ec6f939ba, I recommend using the latest master.
In the coming weeks we should be able to release all the remaining modules we couldn't release yet, in the meantime I recommend using the latest master.
